modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/HorizontalCardTag.java

```diff
@@ -0,0 +1,383 @@
+
+	@Override
+	protected void cleanUp() {
+		super.cleanUp();
+
+		_actionDropdownItems = null;
+		_data = null;
+		_disabled = null;
+		_horizontalCard = null;
+		_href = null;
+		_icon = null;
+		_interactive = null;
+		_selectable = null;
+		_selected = null;
+		_title = null;
+	}
+
+	@Override
+	protected String getHydratedModuleName() {
+		return "frontend-taglib-clay/HorizontalCard";
+	}
+
+	@Override
+	protected Map<String, Object> prepareProps(
+		Map<String, Object> initialProps) {
+
+		Map<String, Object> props = super.prepareProps(initialProps);
+
+		props.put("actions", getActionDropdownItems());
+		props.put("cssClass", processCssClasses(new LinkedHashSet<>()));
+		props.put("disabled", isDisabled());
+		props.put("href", getHref());
+		props.put("icon", getIcon());
+		props.put("id", getId());
+		props.put("interactive", getInteractive());
+		props.put("selectable", isSelectable());
+		props.put("selected", isSelected());
+		props.put("title", getTitle());
+
+		return props;
+	}
+
+	@Override
+	protected String processCssClasses(Set<String> cssClasses) {
+		if (isSelectable()) {
+			cssClasses.add("form-check");
+			cssClasses.add("form-check-card");
+			cssClasses.add("form-check-middle-left");
+		}
+		else {
+			cssClasses.add("card");
+			cssClasses.add("card-horizontal");
+		}
+
+		cssClasses.add("card-type-directory");
+
+		return super.processCssClasses(cssClasses);
+	}
+
+	@Override
+	protected int processStartTag() throws Exception {
+		super.processStartTag();
+
+		Boolean selectable = isSelectable();
+
+		if ((getIcon() == null) && (getTitle() == null) && !selectable) {
+			return EVAL_BODY_INCLUDE;
+		}
+
+		JspWriter jspWriter = pageContext.getOut();
+
+		if (selectable) {
+			jspWriter.write("<div class=\"custom-control custom-checkbox\">");
+			jspWriter.write("<label><input");
+
+			if (isSelected()) {
+				jspWriter.write(" checked");
+			}
+
+			jspWriter.write(" class=\"custom-control-input\" type=\"");
+			jspWriter.write("checkbox\" /><span class=\"custom-control-label");
+			jspWriter.write("\" /><div class=\"card card-horizontal\">");
+
+			_writeDescription(jspWriter);
+
+			jspWriter.write("</div></label></div>");
+		}
+		else {
+			_writeDescription(jspWriter);
+		}
+
+		return SKIP_BODY;
+	}
+
+	private void _writeDescription(JspWriter jspWriter) throws Exception {
+		jspWriter.write("<span class=\"card-body\"><span class=\"card-row\">");
+
+		String icon = getIcon();
+
+		if (icon != null) {
+			jspWriter.write("<div class=\"autofit-col\">");
+
+			StickerTag checkboxTag = new StickerTag();
+
+			checkboxTag.setIcon(icon);
+			checkboxTag.setInline(true);
+
+			checkboxTag.doTag(pageContext);
+
+			jspWriter.write("</div>");
+		}
+
+		String title = getTitle();
+
+		if (title != null) {
+			jspWriter.write("<div class=\"autofit-col autofit-col-expand");
+			jspWriter.write(" autofit-col-gutters\"><p class=\"card-title\"");
+			jspWriter.write(" title=\"");
+
+			String localizedTitle = LanguageUtil.get(
```

Comment:
Why localize in `Tag` class in general? Defensive programming. Some of our tags can have unlocalized attributes, some cannot. It's better to be safe and have an extra localization. Localization of already localized string will not change it.

However, I agree that this is not the best place for it. We should move it to `getTitle()`, like in [AlertTag](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/AlertTag.java#L279-L280).

Reply:
As concluded in conversion, I'll remove the localization.

Reply:
For context, I added a lot of localization in the previous tag conversion pass. It provides enhanced ergonomics and it will be obvious when going through replacement cases.

```
<clay:alert 
    title="warning" 
/>

vs

<clay:alert
    title='<%= LanguageUtil.get(request, "warning") %>'
/>
```

Reply:
@jbalsas So, you are saying we should add internal localization?

/cc @carloslancha @julien

Reply:
I think so, yeah.

This was a very conscious decision in v2 to make the tags as simple as possible and just use them as dumb templates with no magic inside. This was in part influenced by usage of Soy and also our past experience of hiding tons of features behind a seemingly simple surface making them next to imposible to maintain or untangle.

In retrospect, I think we overcompensated a bit too much. In removing all the "magic", we made our tags much harder to use than they needed to be. I feel we should move towards a sensible balance that maximizes our developers productivity and not necessarily their superior knowledge of the platform (for example forcibly understanding where the localization values are coming from). 

In this case, for example, the `LanguageUtil.get...` is also mostly copy-paste code which defeats the purpose of making the APIs simple so developers would be more intentional about what they were doing. Being as is, we can at least make it easier for them not to make some mistakes.

Hope this provides some context as to how I think I went from one side to the other then back to the middle... now, don't take this to mean we should hide all sorts of things and add tons of features under the same component 😉

Reply:
> In retrospect, I think we overcompensated a bit too much.

As v2 predates my time here, I can neither confirm nor refute these claims about [the origin myth](https://en.wikipedia.org/wiki/Origin_myth). 😂